R/getGenomicTiles.R

```diff
@@ -0,0 +1,180 @@
+#' @title Get regions tiling a genome.
+#'
+#' @description Get sequential, potentially annotated regions of a fixed lengths
+#'    (tiles) along chromosomes of a genome.
+#'
+#' @author Michael Stadler
+#'
+#' @param genome The genome to work on. Either a \code{\link[BSgenome]{BSgenome}}
+#'     object, a \code{character} scalar with the name of an installed \code{\link[BSgenome]{BSgenome}}
+#'     or with a file path and name pointing to a fasta file with the genome sequence,
+#'     or a named \code{numeric} vector giving the names and lengths of chromosomes.
+#' @param tileWidth \code{numeric} scalar with the tile length.
+#' @param hasOverlap Named \code{list} with \code{\link[GenomicRanges]{GRanges}} object(s).
+#'     For each list element, a logical vector "X.hasOverlap" will be added to the
+#'     \code{mcols} of the result, with \code{TRUE} for each tile that overlaps
+#'     any region in that element. "X" is obtained from \code{names(hasOverlap)}.
+#' @param fracOverlap  Named \code{list} with \code{\link[GenomicRanges]{GRanges}} object(s).
+#'     For each list element, a numeric vector "X.fracOverlap" will be added to the
+#'     \code{mcols} of the result, with a value between 0 and 1 giving the fraction
+#'     of bases in a tile that overlaps with any region in that element. "X" is
+#'     obtained from \code{names(fracOverlap)}.
+#' @param numOverlap  Named \code{list} with \code{\link[GenomicRanges]{GRanges}} object(s).
+#'     For each list element, two numeric vectors "X.numOverlapWithin" and
+#'     "X.numOverlapAny" will be added to the \code{mcols} of the result, giving
+#'     then number of ranges in that element that are fully contained within
+#'     a tile, or that overlap with a tile in any way, respectively. "X" is
+#'     obtained from \code{names(numOverlap)}.
+#' @param nearest Named \code{list} with \code{\link[GenomicRanges]{GRanges}} object(s).
+#'     For each list element, two numeric vectors "X.nearestName" and
+#'     "X.nearestDistance" will be added to the \code{mcols} of the result, giving
+#'     the name and distance of nearest ranges in that element for each tile. "X" is
+#'     obtained from \code{names(nearest)}, and the values of "X.nearestName" from
+#'     \code{names(nearest$X)}.
+#'
+#' @details The last tile in each chromosome is dropped if it would be shorter
+#'     than \code{tileWidth}. Generated tiles are unstranded (\code{*}) and
```

Comment:
Is there any situation where it would be helpful to know the strand of the overlap? Like for TSSs? If so, and if this information is provided in the `GRanges` object, we could consider including it in the output (one columns per strand?).

Reply:
I guess it would not change that generated tile are unstranded, correct? So, if an annotation source is stranded, it will anyway `ignore.strand` will anyway not matter. Are you suggesting to double all the columns and report overlaps/fractions/numbers separately for annotation elements on each strand? 
If so, could the same also be obtained by splitting the annotation source upfront by strand, and add both seperatly in `getGenomicTiles`?

Reply:
Yes, I like that (the user splitting things in advance) - that makes things easier. (Totally agree that tiles should be unstranded, it would just be a matter of whether or not the strand of the annotations should be taken into account.)